tests/RedisCacheTest.php

```diff
@@ -120,4 +126,73 @@ public function testMultiByteGetAndSet()
         $cache->set($key, $data);
         $this->assertSame($cache->get($key), $data);
     }
+
+    public function testReplica()
+    {
+        $this->resetCacheInstance();
+
+        $cache = $this->getCacheInstance();
+        $cache->enableReplicas = true;
+
+        $key = 'replica-1';
+        $value = 'replica';
+
+        //No Replica listed
+        $this->assertFalse($cache->get($key));
+        $cache->set($key, $value);
+        $this->assertSame($cache->get($key), $value);
+
+        $cache->replicas = [
+            ['hostname' => 'localhost'],
+        ];
+        $this->assertSame($cache->get($key), $value);
+
+        //One Replica listed
+        $this->resetCacheInstance();
+        $cache = $this->getCacheInstance();
+        $cache->enableReplicas = true;
+        $cache->replicas = [
+            ['hostname' => 'localhost'],
+        ];
+        $this->assertFalse($cache->get($key));
+        $cache->set($key, $value);
+        $this->assertSame($cache->get($key), $value);
+
+        //Multiple Replicas listed
+        $this->resetCacheInstance();
+        $cache = $this->getCacheInstance();
+        $cache->enableReplicas = true;
+
+        $cache->replicas = [
+            ['hostname' => '127.0.0.1'],
+            ['hostname' => '127.0.0.1'],
+        ];
+        $this->assertFalse($cache->get($key));
+        $cache->set($key, $value);
+        $this->assertSame($cache->get($key), $value);
+
+        //invalid config
+        $this->resetCacheInstance();
+        $cache = $this->getCacheInstance();
+        $cache->enableReplicas = true;
+
+        $cache->replicas = [
+            ['class' => 'yii\db\Connection'],
```

Comment:
this should expect an exception.

Reply:
My initial intention of not using Instance::ensure() is not to break the application if replica is not properly configured, will update the test to expect exception